src/app/pages/storage2/components/disk-info-card/disk-info-card.component.ts

```diff
@@ -0,0 +1,53 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { TranslateService } from '@ngx-translate/core';
+import { Disk } from 'app/interfaces/storage.interface';
+import { DiskFormComponent } from 'app/pages/storage/disks/disk-form/disk-form.component';
+import { ReplaceDiskDialogComponent, ReplaceDiskDialogData } from 'app/pages/storage/volumes/volume-status/components/replace-disk-dialog/replace-disk-dialog.component';
+import { IxSlideInService } from 'app/services/ix-slide-in.service';
+
+@Component({
+  selector: 'ix-disk-info-card',
+  templateUrl: './disk-info-card.component.html',
+  styleUrls: ['./disk-info-card.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class DiskInfoCardComponent {
+  @Input() disk: Disk;
+  @Input() poolId: number;
+
+  get diskInfo(): { label: string; value: string | number }[] {
```

Comment:
Setting values like this makes it difficult to figure out which values are available while loading. Some values will be available without making a call to middleware. We want to show loading indicators for individual attributes and show the values of attributes which are already available. See the changes made by https://github.com/truenas/webui/pull/6783 for showing loading indicator.